Extension/lib/filter/request-blocking.js

```diff
@@ -71,19 +71,20 @@ WebRequestService.prototype = (function () {
         };
 
         var whitelistRule = this.framesMap.getFrameWhiteListRule(tab);
```

Comment:
Just use the `whitelistRule` and check it for `generichide`, `elemhide` and `jsinject`.
You shouldn't look for another rules anymore.